internal/armada/repository/job.go

```diff
@@ -50,7 +50,7 @@ type JobRepository interface {
 	DeleteJobs(jobs []*api.Job) map[*api.Job]error
 	GetActiveJobIds(queue string, jobSetId string) ([]string, error)
 	GetLeasedJobIds(queue string) ([]string, error)
-	UpdateStartTime(jobId string, clusterId string, startTime time.Time) error
+	UpdateStartTime(jobStartInfos []*JobStartInfo) ([]error, error)
```

Comment:
I think typically we've made the return type something like map[string]error or map[api*Job]error. 
 - Maybe we've deleted most of these. I only see DeleteJobs matching this now

Then its clearer which job id corresponds with this error, rather than relying on index. Although not mandatory, I guess this is more "efficient" than passing as much data around